src/scripts/main.js

```diff
@@ -1,3 +1,40 @@
 'use strict';
 
 // write code here
+const tree = document.querySelector('.tree');
+
+addSpanToFirstTextChild(tree);
+
+tree.addEventListener('click', (e) => {
+  const span = e.target.closest('span');
+  const li = span.closest('li');
+  const list = li.querySelector('ul');
+
+  if (list.style.display === 'none') {
+    list.style.display = '';
+  } else {
+    list.style.display = 'none';
+  }
+});
+
+function addSpanToFirstTextChild(list) {
+  const liList = list.querySelectorAll('li');
+
+  for (const li of liList) {
+    if (!li.firstChild) {
+      return;
+    }
+
+    if (li.firstChild.tagName === 'UL') {
+      addSpanToFirstTextChild(li.firstChild);
+    }
```

Comment:
This condition checks if the first child is a `UL`, but it should also consider cases where the first child is not a text node. Consider adding a check for text nodes.